snapcraft/services/lifecycle.py

```diff
@@ -97,6 +97,23 @@ def get_prime_dir(self, component: str | None = None) -> Path:
                 "because it does not exist."
             ) from err
 
+    @property
+    def prime_dirs(self) -> dict[str | None, Path]:
```

Comment:
This is duplicated for core22 (`snapcraft/parts/parts.py`) and core24 (`snapcraft/services/lifecycle.py`).  There are a handful of duplicated functions between these two locations, so I decided to follow the same paradigm.

Reply:
I de-duplicated it.